tests/test_data_adapter.py

```diff
@@ -418,3 +417,24 @@ def test_cache_vrt(tmpdir, rioda_large):
     cat = DataCatalog(join(root, f"{name}.yml"), cache=True)
     cat.get_rasterdataset(name)
     assert len(glob.glob(join(cat._cache_dir, name, name, "*", "*", "*.tif"))) == 16
+
+
+def test_detect_extent(geodf, geoda, rioda, ts):
```

Comment:
The coverage of the the new methods is not great, see e.g. https://app.codecov.io/gh/Deltares/hydromt/blob/detect-extent/hydromt%2Fdata_adapter%2Fgeodataset.py

Perhaps by creating a small util functions to detect the temporal extend and reuse these between adapters this can be improved.

Reply:
I think coverage is much better now. Can you check that you're happy with it?